src/core/server/saved_objects/migrations/core/build_active_mappings.ts

```diff
@@ -178,6 +178,60 @@ function defaultMapping(): IndexMapping {
       workspaces: {
         type: 'keyword',
       },
+      permissions: {
+        properties: {
+          read: {
```

Comment:
```suggestion
          read: principles
```

as
```
{
   users: { type: 'keyword' },
   groups: { type: 'keyword' },
}
```

are duplicate template codes, A const variable will be better here I think?

Reply:
Not sure if I fully get it, but this is index mapping definition.

Reply:
It is but we can use a variable to get rid of the duplicate code.
```typescript
{ users: { type: 'keyword' }, groups: { type: 'keyword' }, }
```

Reply:
I got your point now, makes sense :)

Reply:
Have changed that.